File: 프로그래머스/unrated/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    x, y = 0, 0
    max_x, max_y = (board[0] - 1) // 2, (board[1] -1) // 2
    for key in keyinput:
        new_x, new_y = x, y
        if key == "up":
            new_y += 1
        elif key == "down":
            new_y -= 1
        elif key == "left":
            new_x -= 1
        elif key == "right":
            new_x += 1
        # 새로운 좌표가 맵의 크기를 벗어나지 않는지 확인
        if -max_x <= new_x <= max_x and -max_y <= new_y <= max_y:
            x, y = new_x, new_y  # 캐릭터의 좌표를 업데이트

    return [x, y]

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0,0};
        for(int i=0;i<keyinput.length;i++){
            switch(keyinput[i]){
                case "up": answer[1]++;
                    break;
                case "down": answer[1]--;
                    break;
                case "left": answer[0]--;
                    break;
                case "right": answer[0]++;
                    break;
            }
            if(Math.abs(answer[0])>board[0]/2){
                answer[0] +=(answer[0]>0) ? -1:1;
            }
            if(Math.abs(answer[1])>board[1]/2){
                answer[1] +=(answer[1]>0) ? -1:1;
            }
            
        }
        return answer;
    }
}